runtime/tests/Test_LightningCoreQIS_Kokkos.cpp

```diff
@@ -0,0 +1,1466 @@
+
+    QUBIT *target = __quantum__rt__qubit_allocate();              // id = 0
+    QirArray *ctrls_arr = __quantum__rt__qubit_allocate_array(1); // id = 1
+
+    QUBIT **ctrls = (QUBIT **)__quantum__rt__array_get_element_ptr_1d(ctrls_arr, 0);
+
+    // qml.Hadamard(wires=0)
+    __quantum__qis__Hadamard(target);
+    // qml.ControlledPhaseShift(0.6, wires=[0,1])
+    __quantum__qis__ControlledPhaseShift(0.6, target, *ctrls);
+    // qml.IsingYY(0.2, wires=[0, 1])
+    __quantum__qis__IsingYY(0.2, target, *ctrls);
+    // qml.CRX(0.4, wires=[1,0])
+    __quantum__qis__CRX(0.4, target, *ctrls);
+
+    constexpr size_t n = 1;
+    constexpr size_t shots = 1000;
+
+    double *buffer = new double[shots * n];
+    MemRefT_double_2d result = {buffer, buffer, 0, {shots, n}, {n, 1}};
+    __quantum__qis__Sample(&result, shots, 1, ctrls[0]);
+    double *samples = result.data_allocated;
+
+    size_t counts0[2] = {0, 0};
+    for (size_t idx = 0; idx < shots * n; idx += n) {
+        if (samples[idx] == 0) {
+            counts0[0]++;
+        }
+        else {
+            counts0[1]++;
+        }
+    }
+
+    CHECK(counts0[0] + counts0[1] == shots);
+
+    QUBIT **qubit = (QUBIT **)__quantum__rt__array_get_element_ptr_1d(ctrls_arr, 0);
+    auto obs = __quantum__qis__NamedObs(ObsId::PauliZ, *qubit);
+
+    CHECK(__quantum__qis__Variance(obs) == Approx(0.0394695).margin(1e-5));
+
+    __quantum__rt__finalize();
+
+    delete[] buffer;
+}
+
+TEST_CASE("Test __quantum__qis__QubitUnitary with an uninitialized matrix",
+          "[qir_lightning_kokkos]")
+{
+    // initialize the simulator
+    char dev[8] = "backend";
+    char dev_value[17] = "lightning.kokkos";
+    __quantum__rt__device((int8_t *)dev, (int8_t *)dev_value);
+    __quantum__rt__initialize();
+
+    QUBIT *target = __quantum__rt__qubit_allocate(); // id = 0
+    MemRefT_CplxT_double_2d *matrix = nullptr;
+
+    REQUIRE_THROWS_WITH(__quantum__qis__QubitUnitary(matrix, 1, target),
+                        Catch::Contains("The QubitUnitary matrix must be initialized"));
+
+    __quantum__rt__finalize();
+}
+
+TEST_CASE("Test __quantum__qis__QubitUnitary with invalid number of wires",
+          "[qir_lightning_kokkos]")
+{
+    // initialize the simulator
+    char dev[8] = "backend";
+    char dev_value[17] = "lightning.kokkos";
+    __quantum__rt__device((int8_t *)dev, (int8_t *)dev_value);
+    __quantum__rt__initialize();
+
+    QUBIT *target = __quantum__rt__qubit_allocate(); // id = 0
+    MemRefT_CplxT_double_2d *matrix = new MemRefT_CplxT_double_2d;
+
+    REQUIRE_THROWS_WITH(__quantum__qis__QubitUnitary(matrix, 3, target),
+                        Catch::Contains("Invalid number of wires"));
+
+    delete matrix;
+
+    __quantum__rt__finalize();
+}
+
+TEST_CASE("Test __quantum__qis__QubitUnitary with invalid matrix", "[qir_lightning_kokkos]")
+{
+    // initialize the simulator
+    char dev[8] = "backend";
+    char dev_value[17] = "lightning.kokkos";
```

Comment:
I'm assuming these tests cannot be parametrized?

Reply:
Ah! I forgot to remove these tests. I was using them to test compiling both compilers, but no need them for this PR and they also can be parameterized.